components/timezone/src/ixdtf.rs

```diff
@@ -248,15 +262,34 @@ impl CustomTimeZone {
     }
 
     #[cfg(feature = "compiled_data")]
-    fn try_from_time_zone_record(record: &TimeZoneRecord<'_>) -> Result<Self, ParseError> {
+    fn try_from_time_zone_record(
+        record: &TimeZoneRecord<'_>,
+        date: &Option<DateRecord>,
+        time: &Option<TimeRecord>,
+    ) -> Result<Self, ParseError> {
+        use icu_calendar::DateTime;
+
         match record {
             TimeZoneRecord::Name(iana_identifier) => {
                 let mapper = TimeZoneIdMapper::new();
                 let bcp47_id = mapper
                     .as_borrowed()
                     .iana_bytes_to_bcp47(iana_identifier)
                     .ok_or(ParseError::InvalidIanaIdentifier)?;
-                Ok(Self::new_with_bcp47_id(bcp47_id))
+
+                let (year, month, day) =
+                    date.map(|d| (d.year, d.month, d.day)).unwrap_or((0, 0, 0));
```

Comment:
If the date or time is not present, then I think do not calculate the metazone. Only calculate it if all the fields are there.